src/Main.elm

```diff
@@ -1,10 +1,251 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+{-| Using elm-live for hot reloading to make development just a bit more comfortable. Overall prefer using create-elm-app because of it's debugging features,
+but that usually requires to generate a brand new project.
+-}
 
+import Browser exposing (element)
+import Color
+import Data.Audience exposing (Audience, allTypes, audiencesJSON, decodeDataAudience)
+import Data.AudienceFolder exposing (AudienceFolder, audienceFoldersJSON, decodeDataAudienceFolder)
+import Html exposing (..)
+import Html.Attributes exposing (class, style)
+import Html.Events exposing (onClick)
+import Json.Decode as Decode exposing (string)
+import List exposing (sortBy)
+import Material.Icons.Outlined as Outlined
+import Material.Icons.Types exposing (Coloring(..))
+import Set exposing (fromList)
 
-main : Html msg
+
+
+{- |
+   currentFolderId - Nothing means root, otherwise (Just folderID) show folder and it's children. The entire model represents the state of the folder tree. The approach I use
+   uses maps and filters to display the folder tree component. It might be better to use Dict for Part 2, as it spares us from having to filter the folder and audience
+   lists every time the category is changed (and only display values stored at a specific key). Classic dict can only use a comparable key though, so no using AudienceType
+   there. Only way would be to match AudienceType to Int values and use this method to initially create the Dict structure. I think presented solution should be satisfactory,
+   as the performance impact should in this case be hardly noticable and we have a guarantee that the view will always work with accurate data (if we dynamically obtained
+   the data, view would always display it accurately - on the opposite site, the Dict would grow outdated and would therefore need to be updated too, which
+   kind of defeats the purpose of creating it in the first place).
+-}
+{- | Use elm-live to launch the app. Install elm live and launch it with commands:
+
+   npm i -g elm-live
+   elm-live src/Main.elm --open -- --output main.js
```

Comment:
Thanks for including the instructions from elm-live, I always forget the required arguments and it is true that `elm reactor` falls too short for many things 👍🏻